core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServicePlugin.java

```diff
@@ -371,6 +374,15 @@ static TypeSignature toTypeSignature(Type type) {
             return TypeSignature.ofList(toTypeSignature(clazz.getComponentType()));
         }
 
+        if (clazz.isEnum()) {
+            return TypeSignature.ofNamed(clazz);
+        }
+
+        final ResponseObject responseObject = AnnotationUtil.findFirst(clazz, ResponseObject.class);
```

Comment:
What do you think of using `StructInfoProvider` to determine whether to convert a return type to a named `TypeSignature`? #4309

Reply:
Did you mean that add entire return type classes to named `TypeSignature` and then filter using `StructInfoProvider`?
e.g. )
```java
    private static NamedTypeInfo newNamedTypeInfo(TypeSignature typeSignature, @Nullable StructInfoProvider structInfoProvider) {
        final Class<?> type = (Class<?>) typeSignature.namedTypeDescriptor();
        if (type == null) {
            throw new IllegalArgumentException("cannot create a named type from: " + typeSignature);
        }

        if (type.isEnum()) {
            return newEnumInfo(type);
        }

        if (structInfoProvider != null) {
            const StructInfo structInfo = structInfoProvider.newStructInfo(type);
            if (structInfo != null) {
                return structInfo;
            }
        }

        return newStructInfo(type);
    }

```

I had similar concerns too about return type to a named `TypeSignature`.

first, I added entire return type classes using `TypeSignature.ofNamed(clazz)`.
But It had simillar problem like #4309. (need to be exclude default class like `HttpResponse`, `Date`, `HttpResult`.)
So I added annotation like `@ResponseObject` and filter it. 😅 

I think It's a good way to solve.

Reply:
OK. I'll fix it contain https://github.com/line/armeria/issues/4309. 😄

Reply:
Unless you want to remove `ResponseObject` and only use the `StructInfoProvider`, I think we can implement it separately in another PR.